spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/AbstractJdbcConfiguration.java

```diff
@@ -106,26 +120,45 @@ public JdbcCustomConversions jdbcCustomConversions() {
 	 */
 	@Bean
 	public JdbcAggregateTemplate jdbcAggregateTemplate(ApplicationEventPublisher publisher,
-			RelationalMappingContext context, JdbcConverter converter, DataAccessStrategy dataAccessStrategy) {
+			RelationalMappingContext context, JdbcConverter converter,
+			ObjectProvider<DataAccessStrategy> dataAccessStrategyProvider, NamedParameterJdbcOperations operations,
+			Optional<NamingStrategy> namingStrategy, @Lazy JdbcConverter jdbcConverter) {
+
+		DataAccessStrategy dataAccessStrategy = dataAccessStrategyProvider.getIfAvailable(() -> dataAccessStrategy( //
+				operations, //
+				jdbcConverter, //
+				jdbcMappingContext(namingStrategy)) //
+		);
 
-		return new JdbcAggregateTemplate(publisher, context, converter, dataAccessStrategy);
+		return new JdbcAggregateTemplate( //
+				publisher, //
+				context, //
+				converter, //
+				dataAccessStrategy //
+		);
 	}
 
 	/**
-	 * Register a {@link DataAccessStrategy} as a bean for reuse in the {@link JdbcAggregateOperations} and the
-	 * {@link RelationalConverter}.
+	 * Create a {@link DataAccessStrategy} for reuse in the {@link JdbcAggregateOperations} and the
+	 * {@link RelationalConverter}. It will return the same instance if called multiple times, regardless of the arguments
+	 * provided. Register a bean of type {@link DataAccessStrategy} if your use case requires a more specialized
+	 * DataAccessStrategy.
 	 *
-	 * @param operations
-	 * @param namingStrategy
-	 * @param jdbcConverter
-	 * @return
+	 * @return Guaranteed to be not {@literal null}.
 	 */
-	@Bean
-	public DataAccessStrategy dataAccessStrategy(NamedParameterJdbcOperations operations,
-			Optional<NamingStrategy> namingStrategy, JdbcConverter jdbcConverter) {
+	private DataAccessStrategy dataAccessStrategy(NamedParameterJdbcOperations operations, JdbcConverter jdbcConverter,
+			JdbcMappingContext context) {
+
+		if (defaultDataAccessStrategy == null) {
 
-		JdbcMappingContext context = jdbcMappingContext(namingStrategy);
-		return new DefaultDataAccessStrategy(new SqlGeneratorSource(context), context, jdbcConverter, operations);
+			defaultDataAccessStrategy = new DefaultDataAccessStrategy( //
+					new SqlGeneratorSource(context), //
```

Comment:
Maybe extract into a local variable to keep the line shorter and avoid the additional manual breaks.